src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -670,28 +670,28 @@ private ValueGetter<VBuffer<uint>> ComposeGetterVec(IRow input, int iinfo)
 
         #region Core Hash functions, with and without index
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private static uint HashCore(uint seed, ref DvText value, uint mask)
+        private static uint HashCore(uint seed, ref ReadOnlyMemory<char> value, uint mask)
         {
             Contracts.Assert(Utils.IsPowerOfTwo(mask + 1));
-            if (!value.HasChars)
+            if (value.IsEmpty)
                 return 0;
-            return (value.Trim().Hash(seed) & mask) + 1;
```

Comment:
>Trim() [](start = 26, length = 6)

We used to call `DvText.Trim()` here, which only trims space ' ' characters. Now we are calling ROS.Trim(), which trims all whitespace. My initial calculations say changing this to ROMUtils.TrimSpaces takes off ~2%.